### DISCUSSION TITLE
Wyze Cam v3 power state

### DISCUSSION THREAD

**Discussion by benjani85** [mrlt8/docker-wyze-bridge#359] [Q&A]
Just wondering if there is any way to determine whether the camera is powered on or off? I initially thought this could be achieved using the state of idle/unavailable however I don't believe this to be the right way to check the power state as the camera often shows as idle when it is off.

**Comment by mrlt8** [comment]
[There is](https://github.com/mrlt8/docker-wyze-bridge/blob/fa2dc85ec4e6aadf0423a6b9dd4fce14723e8015/app/wyzecam/tutk/tutk.py#L865), but I'm not sure how reliable it is. We used it at one point, but I think it was easier to just keep attempting the connection. May revisit this once we get the audio branch merged.


In the meantime, have you looked into one of the web-based APIs like [wyze-sdk](https://github.com/shauntarves/wyze-sdk) or [ha-wyzeapi](https://github.com/JoshuaMulliken/ha-wyzeapi)?

**Reply by benjani85** [reply]
Is there anyway to use the logs of Wyze Bridge to get the status of the camera and turn that into a sensor for on/off? It seems the logs already know the camera state.

**Reply by mrlt8** [reply]
We're already catching the error code from the tutk library, so we could probably use that to push an mqtt message or do a webhook.

e.g. offline is usually `-90`: 
```c
/** All Server response can not find device */
#define IOTC_ER_DEVICE_OFFLINE						-90 
```

**Reply by benjani85** [reply]
Sorry I'm quite new to home assistant so not sure how to go about this.

**Reply by benjani85** [reply]
I ended up working around this by monitoring the timestamp of the snapshot image file and comparing it against the current time to determine if it has been recently updated or not.

**Reply by mrlt8** [reply]
Pushed some [changes](https://github.com/mrlt8/docker-wyze-bridge/commit/296255b63371e9c88a570027e52868ffcadc8855) to the dev branch so the bridge should push the more info to mqtt:
<img width="308" alt="state" src="https://user-images.githubusercontent.com/67088095/167260709-54b443b5-4201-48c2-a8a5-3fdcd96bcd14.png">
<img width="248" alt="offline" src="https://user-images.githubusercontent.com/67088095/167260712-67d1a930-c953-438d-82d5-d65034044b3d.png">
Also thinking of adding webhooks via something like IFTTT so that we can get notifications when the cam goes "-90".

Would appreciate feedback and suggestions!

**Reply by benjani85** [reply]
I saw that the power state has been added to mqtt support now! I'm trying to configure this however I'm unable to get anything to show up in Mosquitto Broker. It is configured with the default core-mosquitto broker and port including a username/password.

In the MQTT fields I have used format IP:Port
For credentials username:password
I haven't filled mqtt_topic and mqtt_dtopic was pre-populated with "homeassistant"

Is there anything I'm missing?

**Reply by mrlt8** [reply]
hmm, I believe the default bridge config should pull the MQTT host/port and user/pass from `supervisor/services/mqtt`, but it should also work if manually configured.

Do you get any error in the log?

**Reply by benjani85** [reply]
There is nothing in the log after the addon is restarted

**Reply by mrlt8** [reply]
hmm, and nothing on `wyzebridge/#`?

**Reply by benjani85** [reply]
if you mean under config/wyze-bridge there are just the usual files. or do you mean something else?

**Reply by mrlt8** [reply]
In the MQTT topic.

**Reply by benjani85** [reply]
I don't see any devices for wyzebridge under the MQTT integration

**Reply by mrlt8** [reply]
I believe it should get pushed as an entity with the mqtt_dtopic.

**Reply by benjani85** [reply]
I managed to see it using wyzebridge/# then created a sensor in the configuration file for the topic. Thanks for the help!

**Comment by mrlt8** [comment]
Can you try deleting all the other MQTT options in your bridge env and just leave `mqtt_dtopic` to homeassistant? You should then be able to see the messages get sent to the topics under  `wyzebridge/#`

**Reply by benjani85** [reply]
Ok after that I can see wyzebridge entries if I try listening to the topic in mqtt. But how to I get it to appear as an entity?
